flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/TableFunction.scala

```diff
@@ -135,8 +135,11 @@ abstract class TableFunction[T] extends UserDefinedFunction {
     *
     * @param arguments arguments of a function call (only literal arguments
     *                  are passed, nulls for non-literal ones)
+    * @param typeInfos The type information of the parameters. only valid argument types
```

Comment:
Yes, I think we should resolve the expressions one more time to get all the TypeInformations.